src/common/weighted_shuffle.h

```diff
@@ -0,0 +1,41 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * This is free software; you can redistribute it and/or
```

Comment:
no need to repeat the license. see https://github.com/ceph/ceph/blob/master/CodingStyle#L73